### ISSUE TITLE
Why is every file that matches the config extension used as an entry?

### ISSUE THREAD

**Issue by scottcorgan** [rails/webpacker#599]
While investigated slow build times in our app, I found that every single file that matches the extensions setting in `webpacker.yml` is set to be an entry file. I might be wrong here, but that may not be necessary.

In my testing, I've found only setting the files that use `javascript_pack_tag` or `stylesheet_pack_tag` are needed to properly build and compile all the types of static assets.

I'm super curious about this and want to know if anyone has any insight.

Thanks! 👍

**Comment by Odaeus** [comment]
I'm still learning Webpack and noticed this too. The webpacker README [seems](https://github.com/rails/webpacker#paths) to [indicate](https://github.com/rails/webpacker#app-structure) that only the files in the root of the `packs` dir will be entry points. However, I think that all source files are being built and emitted as separate possible entry points.

I modified the `extensionGlob` in [shared.js](https://github.com/rails/webpacker/blob/master/lib/install/config/webpack/shared.js#L14) to remove the recursive search (`"**/*{$...` -> `"*{$...`) and now my entry points are just the top level files.

Though it didn't seem to make much difference to compilation time!

**Comment by gauravtiwari** [comment]
@scottcorgan Only the files that are put under packs folder are matched. Please make sure to remove other source file out of the packs folder to `app/javascript` and only keep entry files inside `packs` folder. 

```yml
app/javascript: 
  - calender.js
  - chat.js
  - src/components/chat.js
  - # etc....
packs: 
  - main.js # imports child components and bootstrap
```

**Comment by gauravtiwari** [comment]
@Odaeus Yes that doesn't add anything however please make sure to not have all the JS files inside packs folder - only entry files needs to be here.

**Comment by Odaeus** [comment]
Thanks @gauravtiwari, I guess I wasn't paying close enough attention to the README, oops!

**Comment by scottcorgan** [comment]
Is there something in the README you could link to about this?

**Comment by gauravtiwari** [comment]
@scottcorgan I will make a PR and add a section for this 👍

**Comment by adamsanderson** [comment]
Hey thanks, that tripped me up too!

**Comment by gauravtiwari** [comment]
Addressed in #710
